### PR TITLE
Report detailed errors on task creation

### PR DESCRIPTION
**Pull request by jaragunde**
This is where I wanted to go with all of this :slightly_smiling_face:. One of the most annoying aspects of phpreport is the lack of feedback on error, and creating/updating/deleting tasks the most used features, by far, so this change can have a big, positive impact.

I consider this WIP.

### CODE REVIEW

**Comment by jaragunde** [comment]
I wanted some feedback because I found some "tension" here, when returning errors. Some errors are individual of the tasks, like when we try to assign a task to a closed project, and some are collective, like when we try to create two tasks that overlap with each other.

I started seeing this from the collective point of view (overlapping tasks) so I made the operation return one single OperationResult instead of an array of them, like we do in the corresponding projects. Now I'm having second thoughts... Should I change and return one OperationResult per created task? What should I do when there are overlapping tasks, maybe return multiple, equal OperationResults?

**Comment by jaragunde** [comment]
After discussing to myself, I resolved to return an array of OperationResults from batchCreate(), and in case of overlapping I return an array containing one result. If we ever need to match tasks with their result for better error handling/reporting, we can modify this behavior.

**Review by anarute — APPROVED** [review]
Apart from the nit it's looking good to me. I haven't had time to test it though, I'll do it soon.

File: model/facade/action/CreateTasksAction.php [hunk 1]
```diff
@@ -68,40 +68,39 @@ public function __construct($tasks) {
...
 
         //TODO: do something meaningful with the list of discarded tasks
-        if (empty($discardedTasks)) {
-            return 0;
-        }
-        else {
-            return -1;
+        if (!empty($discardedTasks)) {
+            $result = new OperationResult(false);
+            $result->setMessage("Some tasks were discarded:\n" . $discardedReason);
```
[the remark this answers is not in the document]
Comment (dmtrek14) [reply model/facade/action/CreateTasksAction.php]: Agree.

**Review by dmtrek14 — APPROVED** [review]
Other than the small wording change that Ana recommended, this looks great. It's nice to start seeing the more informative errors in more places - especially task creation since that's probably the most common action performed by users.

File: model/facade/action/CreateTasksAction.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaragunde) [reply model/facade/action/CreateTasksAction.php]: Done. Thanks!

**Pull request merged**